src/compiler/utilities.ts

```diff
@@ -10492,7 +10493,8 @@ function isIdentifierInNonEmittingHeritageClause(node: Node): boolean {
                 return "quit";
         }
     }) as HeritageClause | undefined;
-    return heritageClause?.token === SyntaxKind.ImplementsKeyword || heritageClause?.parent.kind === SyntaxKind.InterfaceDeclaration;
+    if (heritageClause === undefined) return false;
+    return isJSDocImplementsTag(heritageClause) || heritageClause.token === SyntaxKind.ImplementsKeyword || heritageClause.parent.kind === SyntaxKind.InterfaceDeclaration;

```

Comment:
`isIdentifierInNonEmittingHeritageClause` is only used for `isValidTypeOnlyAliasUseSite`. Is is the case we could instead make `isValidTypeOnlyAliasUseSite` just _always_ allow use from JSDoc? After all, those are necessarily non-JS-code-emitting nodes.

Reply:
@jakebailey I’ve updated it to _always_ allow usage from JSDoc and removed the specific case with the implements tag. Thanks